File: app/src/main/java/fr/android/androidexercises/BookItemView.java
package fr.android.androidexercises;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;

public class BookItemView extends LinearLayout {

    private TextView nameTextView;
    private TextView priceTextView;
    private ImageView coverImageView;

    public BookItemView(Context context) {
        this(context, null);
    }

    public BookItemView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public BookItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        nameTextView = (TextView) findViewById(R.id.titleTextView);
        priceTextView = (TextView) findViewById(R.id.priceTextView);
        coverImageView = (ImageView) findViewById(R.id.coverImageView);
    }

    public void bindView(Book book) {
        nameTextView.setText(book.getTitle());
        priceTextView.setText(String.valueOf(book.getPrice()));
        Glide.with(this)
                .load(book.getCover())
                .into(coverImageView);
    }
}

File: app/src/main/java/fr/android/androidexercises/BookItemView.kt
package fr.android.androidexercises

import android.widget.TextView
import android.widget.LinearLayout
import android.content.Context;
import android.util.AttributeSet;



/**
 * Created by Lumi on 10/11/2017.
 */

class BookItemView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : LinearLayout(context, attrs, defStyleAttr) {

    private var nameTextView: TextView? = null
    private var priceTextView: TextView? = null

    override fun onFinishInflate() {
        super.onFinishInflate()
        nameTextView = findViewById(R.id.nameTextView)
        priceTextView = findViewById(R.id.priceTextView)
    }

    fun bindView(book: Book) {
        nameTextView!!.text = book.name
        priceTextView!!.text = book.price.toString()
    }
}